test/ILLink.RoslynAnalyzer.Tests/TestChecker.cs

```diff
@@ -67,18 +73,39 @@ bool IsExpectedDiagnostic (AttributeSyntax attribute)
 			switch (attribute.Name.ToString ()) {
 			case "ExpectedWarning":
 				var args = TestCaseUtils.GetAttributeArguments (attribute);
-				if (args.TryGetValue ("ProducedBy", out var producedBy) &&
-					producedBy is MemberAccessExpressionSyntax memberAccessExpression &&
-					memberAccessExpression.Name is IdentifierNameSyntax identifierNameSyntax &&
-					identifierNameSyntax.Identifier.ValueText == "Trimmer")
-					return false;
+				if (args.TryGetValue ("ProducedBy", out var producedBy)) {
+					var producedByValues = GetProducedByValues (producedBy);
+					// Skip if this warning is not expected to be produced by any of the analyzers that we are currently testing.
+					return !producedByValues.Contains ("Analyzer") && (producedByValues.Intersect (TestingAnalyzers) == null);
+				}
+
 				return true;
 			case "LogContains":
 			case "UnrecognizedReflectionAccessPattern":
 				return true;
 			default:
 				return false;
 			}
+
+			static List<string> GetProducedByValues (ExpressionSyntax expression)
```

Comment:
I wonder - should we make this return a `ProducedBy` value? Yes, we would have to map it back into strings to compare with the set of active analyzers - but I would find it a little easier to reason about since it would collapse arguments like `ProducedBy.RUCAnalyzer | ProducedBy.Analyzer` into `ProducedBy.Analyzer`. Also ideally we would handle `ProducedBy.TrimmerAndAnalyzer` if passed explicitly.